### DISCUSSION TITLE
How to scan directories on production on Vercel (with Next.js)?

### DISCUSSION THREAD

**Discussion by alexandrepaiva-dev** [vercel/community#178] [Help] [answered]
Hi,

I've tried to ask for help on Stackoverflow already but no answers after 10 days, also tried to contact Vercel but no reply. Hopefully I will find someone to help me here.

I'm developing a sitemap generator sript for my website, it works well on dev (yarn dev) and production (yarn build & yarn start) on my own PC but when uploading to Vercel I get an error and I can't find the solution. My website has the following structure:

https://xxxxx.com/
https://xxxxx.com/projects
https://xxxxx.com/projects/one
https://xxxxx.com/projects/two
https://xxxxx.com/projects/three
https://xxxxx.com/contact
etc

The way I access the sitemap is https://xxxxx.com/sitemap.xml (which the file is placed under "/pages" as "sitemap.xml.js"), again, it works when I access on my own PC either on dev or build/start.

But on Vercel I get the error:

`ERROR   Error: ENOENT: no such file or directory, scandir '/var/task/.next/server/pages/projects'`

More details on Stackoverflow:
https://stackoverflow.com/questions/70027742/how-to-scan-directories-on-production-on-vercel-with-next-js

It seems like it cannot find the directory "projects" under "/.next/server/pages". 

Does anyone know how to properly scan files/directories when in production on Vercel?

Thank you!

**Comment by Pieparker** [comment, marked as answer]
This conversation appears to have gone quiet and I can see that we responded to your private case where we received no response–I hope you were able to overcome this issue!

To keep discussions on topic and to prevent the resurrection of old topics, this discussion has been closed. If issues persist or a future reader stumbles upon this discussion and is exhibiting the same problem, please create a new discussion.
